packages/engine-cli/src/engine-build.ts

```diff
@@ -377,10 +383,15 @@ function runOnDemandSingleEnvironment(
         return runningNodeManager.port;
     }
     function middleware(req: express.Request, res: express.Response) {
-        console.log(
-            `running on demand feature: "${req.body.featureName}" config: "${req.body.configName}" runtimeArgs: "${req.body.runtimeArgs}"`,
-        );
-        run(req.body.featureName, req.body.configName, req.body.runtimeArgs)
+        let message = `running on demand feature: "${req.body.featureName}" config: "${req.body.configName}"`;
+        if (req.body.runtimeArgs) {
+            message += ` runtimeArgs: "${req.body.runtimeArgs}"`;
+        }
+        if (req.body.restart) {
+            message += ' with restart';
+        }
+        console.log(message);
+        run(req.body.featureName, req.body.configName, req.body.runtimeArgs, req.body.restart)
```

Comment:
restart can be undefined

Reply:
Yep will fix

Reply:
It does not really matter because we only care about the truth case. I will consider removing it and always wait